Std/Classes/RatCast.lean

```diff
@@ -0,0 +1,23 @@
+/-
+Copyright (c) 2014 Robert Lewis. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robert Lewis, Leonardo de Moura, Johannes Hölzl, Mario Carneiro, Gabriel Ebner
+-/
+import Std.Data.Rat.Basic
+
+/-- Type class for the canonical homomorphism `Rat → K`. -/
+class RatCast (K : Type u) where
+  /-- The canonical homomorphism `Rat → K`. -/
+  protected ratCast : Rat → K
+
+/-- Canonical homomorphism from `Rat` to a division ring `K`.
+This is just the bare function in order to aid in creating instances of `DivisionRing`. -/
+@[coe, match_pattern] protected def Rat.cast {K : Type u} [RatCast K] : Rat → K := RatCast.ratCast
```

Comment:
:+1: on making this uniform with {Nat,Int}.cast.

But this is now inconsistent with mathlib3.  Do we want to backport this change, or do you think we can patch this over using `#align`?

Reply:
What's the inconsistency? Is it that there is no `RatCast` class in mathlib3? It seems relatively unlikely to break anything but a backport is the easiest way to be sure and avoid periodic questions from porters about it.

Reply:
There is an analogous `RatCast` class in mathlib; the current alignment is `#align has_rat_cast RatCast` in Mathlib.Algebra.Field.Defs. (As far as I can see there's no alignment for its field `rat_cast`.)

Reply:
The difference to the existing `RatCast` class is that we don't have a helper `Rat.cast` definition but call `RatCast.ratCast` directly right now.  That's why I was concerned about the alignment because we're mapping 1 mathlib3 definition to 2 mathlib4 ones.

Reply:
I see, makes sense—should we e.g. prefer `Rat.cast` everywhere in mathlib4 then?

When we bump mathlib4's version of std4 after this PR, we could(?) `#align has_rat_cast.rat_cast Rat.cast` and do a find-replace in mathlib4 for `RatCast.ratCast` → `Rat.cast` (of which there aren't many occurrences right now, so we could make sure this really is the right thing to do in each of them). (If this PR is merged, we'll need to do that to fix at least one `norm_cast` lemma anyway, since a different thing has `@[coe]` now.)

Not sure of any subtle gotchas waiting for us here, though...